Do not use the ancient flat namespace. Use the two-level namespace which has
been the default since Mac OS X 10.1.
--- configure.in.orig	2019-12-28 08:19:24.000000000 -0600
+++ configure.in	2024-04-29 02:28:42.000000000 -0500
@@ -333,7 +333,7 @@
 	if test x"$macosx_module" = x"no"; then
 	   AC_MSG_RESULT([dylib options])
 	   SHARED_LIB="libxvidcore.\$(API_MAJOR).\$(SHARED_EXTENSION)"
-	   SPECIFIC_LDFLAGS="-Wl,-read_only_relocs,suppress -dynamiclib -flat_namespace -compatibility_version \$(API_MAJOR) -current_version \$(API_MAJOR).\$(API_MINOR) -install_name \$(libdir)/\$(SHARED_LIB)"
+	   SPECIFIC_LDFLAGS="-Wl,-read_only_relocs,suppress -dynamiclib -compatibility_version \$(API_MAJOR) -current_version \$(API_MAJOR).\$(API_MINOR) -install_name \$(libdir)/\$(SHARED_LIB)"
 	else 
 	   AC_MSG_RESULT([module options])
  	   PRE_SHARED_LIB="libxvidcore.\$(SHARED_EXTENSION)-temp.o"
